src/main/java/com/cs98/VerbatimBackend/controller/AccountSettingsController.java

```diff
@@ -0,0 +1,84 @@
+package com.cs98.VerbatimBackend.controller;
+
+import com.cs98.VerbatimBackend.model.User;
+import com.cs98.VerbatimBackend.repository.UserRepository;
+import com.cs98.VerbatimBackend.request.RegisterRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.ObjectUtils;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import com.cs98.VerbatimBackend.request.AccountSettingsRequest;
+
+import java.util.Optional;
+
+@RestController
+public class AccountSettingsController {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @PostMapping(path = "api/v1/accountSettings")
+    public ResponseEntity<User> register(@RequestBody AccountSettingsRequest request) {
+        User user;
+
+        // if user does not exist, return bad request
+        if (ObjectUtils.isEmpty(request.getUsername())) {
+            return ResponseEntity.badRequest().build();
+        }
+        // if the user exists, update changed fields
+        else if (userRepository.existsByUsername(request.getUsername())) {
+            user = userRepository.findByUsername(request.getUsername());
+
+            if (request.getNewUsername() != null &&                            // make sure username is not null
+                    !request.getNewUsername().isEmpty() &&                     // make sure username is not empty
+                    !request.getNewUsername().equals(user.getUsername())) {    // don't update if not changed
+                Optional<User> userOptional = Optional.ofNullable(userRepository.
+                        findByUsername(request.getNewUsername()));
+                if (userOptional.isPresent()) {                             // check if username is taken
+                    throw new IllegalStateException("username taken");
```

Comment:
Should probably return ResponseEntity.badRequest here so that frontend knows to notify user that their new username was taken

Reply:
Good idea. I'll make that fix.